pkg/proxy/ipvs/proxier.go

```diff
@@ -1116,7 +1116,7 @@ func (proxier *Proxier) syncProxyRules() {
 				// There is no need to bind Node IP to dummy interface, so set parameter `bindAddr` to `false`.
 				if err := proxier.syncService(svcNameString, serv, false); err == nil {
 					activeIPVSServices[serv.String()] = true
-					if err := proxier.syncEndpoint(svcName, false, serv); err != nil {
+					if err := proxier.syncEndpoint(svcName, svcInfo.OnlyNodeLocalEndpoints, serv); err != nil {
```

Comment:
The 2nd parameter of `proxier.syncEndpoint()` is for indicating if we bind service address to dummy interface. I am not sure the change can fix your problems.